Asterix/optics/propagation_functions.py

```diff
@@ -610,30 +615,81 @@ def prop_fpm_regional_sampling(pup,
     EF_before_LS : 2D array (complex)
         E-field before the Lyot stop.
     """
-    from Asterix.optics import shift_phase_ramp
+    samp_outer = 2
+
+    if not isinstance(nbres, (list, np.ndarray)):
+        raise TypeError(f"'nbres' parameter need to be of type list or numpy.array. Currently type = {type(nbres)}")
 
-    if samp_outer != 2:
-        raise ValueError(f"samp_outer' needs to be 2, otherwise we cut off the high-spatial frequencies and the"
-                         f"simulation turns out bad. This can become a variable parameter in the future if we allow"
-                         f"the array sizes to be adapted in this function as well, in order to retain the original"
-                         f"spatial frequency content in the outer sampling layer.")
+    nbres = np.array(nbres)
 
-    dim_pup = pup.shape[0]
+    dim_overpad_pupil = pup.shape[0]
     dim_fpm = fpm.shape[0]
-    pup0 = np.array(pup, copy=True, dtype='complex128')
+
+    if real_dim_input is None:
+        real_dim_input = pup.shape[0]
+    samplings = real_dim_input / nbres
+
+    if not np.all(np.diff(nbres) >= 0):  # check if it is sorted by checking 2 by 2 diff is always positive
```

Comment:
If possible, please delete this comment, it is just a less cleared repeat of the documentation for `numpy.diff` and almost confusing here.

Reply:
done